### PR TITLE
Akamai purger updated as per the latest API

### PR DESCRIPTION
**Pull request by neetumorwani**

### CODE REVIEW

**Comment by neetumorwani** [comment]
@cam8001 Can we please review and merge this. PR has nitpicks I noticed while testing.

File: src/AkamaiClient.php [hunk 1]
```diff
@@ -426,8 +426,10 @@ public function removeInvalidUrls($urls) {
     $urls_to_clear = [];
     $base_url = $this->drupalConfig->get('basepath');
     foreach ($urls as $path) {
-      if ($path[0] === '/') {
-        $path = ltrim($path, '/');
+      if (is_array($path)) {
```
Comment (cam8001) [inline src/AkamaiClient.php]: I don't understand this change? Aren't we iterating through an array of strings, so `$path` should be a string?

File: src/Plugin/Purge/Purger/AkamaiPurger.php [hunk 2]
```diff
@@ -93,4 +85,11 @@ public function invalidate(array $invalidations) {
     $this->client->purgeUrls($urls_to_clear);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function hasRuntimeMeasurement() {
+    return TRUE;
+  }
+
```
Comment (cam8001) [inline src/Plugin/Purge/Purger/AkamaiPurger.php:150]: I don't really understand this API change, I guess we need to feed back runtime stats to Purge somehow over the course of the Purge cycle?

Comment (cam8001) [inline src/Plugin/Purge/Purger/AkamaiPurger.php]: Ok, so looking at Purge, it appears that if we report this as true we need to implement a RuntimeMeasurementInterface.

**Comment by cam8001** [comment]
Can we add in status reporting to `invalidate`? At present, as we are not returning a timehint, AND we never move items out of the processing state.

File: src/Plugin/Purge/Purger/AkamaiPurger.php [hunk 3]
```diff
@@ -82,7 +82,10 @@ public function invalidate(array $invalidations) {
       }
     }
 
-    $this->client->purgeUrls($urls_to_clear);
+    $response = $this->client->purgeUrls($urls_to_clear);
+    if($response->getStatusCode() == 201) {
+      $invalidation->setState(InvalidationInterface::SUCCEEDED);
```
Comment (cam8001) [inline src/Plugin/Purge/Purger/AkamaiPurger.php]: I don't think this code will ever run as its outside of the foreach

File: src/Plugin/Purge/Purger/AkamaiPurger.php (hunk 2 above)
Comment (nielsvm) [inline src/Plugin/Purge/Purger/AkamaiPurger.php:150]: No, Purge takes care of all of this for you actually. Just by returning TRUE, its saying "yes, do this runtime tracking please"

**Pull request closed without merging**